pkg/slo-controller/noderesource/plugins/midresource/plugin.go

```diff
@@ -196,3 +246,26 @@ func prepareNodeForResource(node *corev1.Node, nr *framework.NodeResource, name
 		node.Status.Allocatable[name] = *q
 	}
 }
+
+func getPercentFromStrategy(strategy, defaultStrategy *configuration.ColocationStrategy, strategyType string) float64 {
+	switch strategyType {
+	case MidCPUThreshold:
+		if strategy == nil || strategy.MidCPUThresholdPercent == nil {
+			return float64(*defaultStrategy.MidCPUThresholdPercent) / 100
+		}
+		return float64(*strategy.MidCPUThresholdPercent) / 100
+	case MidMemoryThreshold:
+		if strategy == nil || strategy.MidMemoryThresholdPercent == nil {
+			return float64(*defaultStrategy.MidMemoryThresholdPercent) / 100
+		}
+		return float64(*strategy.MidMemoryThresholdPercent) / 100
+	case MidUnallocatedPercent:
+		if strategy == nil || strategy.MidUnallocatedPercent == nil {
+			return float64(*defaultStrategy.MidUnallocatedPercent) / 100
+		}
+		return float64(*strategy.MidUnallocatedPercent) / 100
+	default:
+		// fixme: what about default?
+		return -1
```

Comment:
for the current usage, it better return zero

Reply:
fixed.